manticore/platforms/evm.py

```diff
@@ -0,0 +1,1760 @@
+'''
+Solidity / Smart contract VM
+Implements the yellow paper: http://gavwood.com/paper.pdf
+Get example contracts from here:
+https://ethereum.github.io/browser-solidity/#version=soljson-latest.js
+'''
+import random, copy
+from ..platforms.platform import *
+from ..core.smtlib import solver, TooManySolutions, Expression, Bool, BitVec, Array, Operators, Constant, BitVecConstant, ConstraintSet
+from ..core.state import ForkState, TerminateState
+from ..utils.helpers import issymbolic
+from ..utils.event import Eventful
+from ..core.smtlib.visitors import pretty_print, arithmetic_simplifier, translate_to_smtlib
+from ..core.state import Concretize,TerminateState
+import logging
+import sys, hashlib
+if sys.version_info < (3, 6):
+    import sha3
+
+logger = logging.getLogger('PLATFORM')
+
+TT256 = 2 ** 256
+TT256M1 = 2 ** 256 - 1
+TT255 = 2 ** 255
+TOOHIGHMEM = 0x100
+
+def to_signed(i):
+    return i if i < TT255 else i - TT256
+
+def pack(i, size=32):
+    assert size >=1
+    o = [0] * size
+    for x in range(size):
+        o[(size-1) - x] = i & 0xff
+        i >>= 8
+    return ''.join(map(chr,o))
+
+class EVMMemory(object):
```

Comment:
Probably not in this PR, but should we eventually make our regular `Memory` be more generic so we can use it as a base class for arbitrary VMs?

Reply:
Agree. We should explore the notion. Though there are differences.  LinuxMemory has mmap and EVMMemory is compacted fro 0 to a variable max. 
We should document our memory backends everywhere and even provide alternatives like:
* Eager+ITE  <- we are using this and its undocumented 
* Lazy+Array